### ISSUE TITLE
Add stratify=y to train_test_split and balanced_accuracy_score to RF metrics

### ISSUE THREAD

**Issue by turbomam** [contextualizer-ai/env-embeddings#73]
## Summary

The Random Forest training currently lacks two important sklearn best practices identified by PIs:
1. **Stratified train/test split** - ensures proportional class representation
2. **Balanced accuracy metric** - complements our existing weighted F1 reporting

## Background

From PI discussion (Marcin @ 1:38 PM, 4:59 PM):
> you could add a stratify=y which then will split the classes proportionally across train/test. that is likely to improve performance actually. I think in this setting we want all classes represented in training because otherwise can't predict on them.

> The weighting will be frequency of each class … [and we should report] balanced_accuracy_score

## Current State

**rf_analysis.py:478-480** - Missing stratify parameter:
```python
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=test_size, random_state=RANDOM_STATE
)  # ❌ Missing stratify=y!
```

**rf_analysis.py:200-213** - Reports weighted F1 but not balanced accuracy:
```python
train_report = classification_report(...)
test_report = classification_report(...)
train_f1 = train_report["weighted avg"]["f1-score"]
test_f1 = test_report["weighted avg"]["f1-score"]
# ❌ Missing balanced_accuracy_score
```

### 1. Add stratify=y to train_test_split

**File:** `src/env_embeddings/rf_analysis.py:478-480`

```python
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=test_size, random_state=RANDOM_STATE, stratify=y
)
```

**Benefits:**
- Ensures rare classes appear in both train and test sets
- Prevents random variance from skipping classes in test set
- Critical for NCBI dataset with 500+ classes
- Improves reproducibility and fairness of evaluation

### 2. Add balanced_accuracy_score to metrics

**File:** `src/env_embeddings/rf_analysis.py` (train_rf_model function)

Add import:
```python
from sklearn.metrics import accuracy_score, classification_report, balanced_accuracy_score
```

Add metric computation after line 213:
```python
train_balanced_acc = balanced_accuracy_score(y_train, y_train_pred)
test_balanced_acc = balanced_accuracy_score(y_test, y_test_pred)
```

Return in dict (line 236+):
```python
return {
    "model": rf,
    "train_f1": train_f1,
    "test_f1": test_f1,
    "train_accuracy": train_acc,
    "test_accuracy": test_acc,
    "train_balanced_accuracy": train_balanced_acc,  # NEW
    "test_balanced_accuracy": test_balanced_acc,    # NEW
    "cv_mean": cv_scores.mean(),
    ...
}
```

Update print statement (line 487):
```python
print(
    f"    Classes: {result['n_classes']:3d} | "
    f"Test F1: {result['test_f1']:.3f} | "
    f"Balanced Acc: {result['test_balanced_accuracy']:.3f} | "  # NEW
    f"CV: {result['cv_mean']:.3f}±{result['cv_std'] * 2:.3f} | "
    f"Overfit: {result['overfitting']:+.3f}"
)
```

## Implementation Difficulty

**Very Easy** - ~10 lines of code changes across 3 locations in one file.

## Testing

Should verify:
1. Stratified split works even with singleton classes (may need conditional logic)
2. Balanced accuracy is reported correctly in all outputs
3. Existing tests pass
4. Re-run RF analysis notebooks to compare before/after metrics

## Related Work

- ✅ Already using `class_weight='balanced'` (commit 1a27352)
- ✅ Already implemented hierarchical scoring with is_a hops (commit a017c54)
- 🔮 Future: Information Content (IC) based scoring (see PI discussion, lower priority)

## References

- PI Discussion: 2024-10-16 Slack thread
- Commit 1a27352: Added class_weight='balanced' 
- scikit-learn docs: https://scikit-learn.org/stable/modules/model_evaluation.html#balanced-accuracy-score

---

**Priority:** High - Core evaluation metric that should be fixed before further analysis
**Effort:** Small - ~30 minutes including testing
**Impact:** Medium-High - Better class representation and more complete metrics reporting
